fvm/src/machine/mod.rs

```diff
@@ -124,8 +124,8 @@ impl NetworkConfig {
     pub fn new(network_version: NetworkVersion) -> Self {
         NetworkConfig {
             network_version,
-            max_call_depth: 4096,
-            max_wasm_stack: 64 * 1024,
+            max_call_depth: 1024,
```

Comment:
To be technically correct, we should set 1024 on nv16 and above, only.

Reply:
So, @magik6k pointed out that we'll hit the rust stack limit at 4096. The only real solution would be to increase the thread size.

But, in practice, it's unclear if one can actually _make_ a message that can get past 1024 calls on mainnet. By "unclear", I mean "totally not close to possible". @magik6k  only got to 1025 by making a custom actor that did nothing but recursively send. On mainnet, one would have to make a recursive call in a multisig actor, where each recursive call would be significantly more expensive (running out of block gas before anything else).

Reply:
Tradeoffs:

- The current way, it's unlikely that something goes wrong, and the worst-case is a fork (for anyone running with the FVM_EXPERIMENTAL flag).
- By increasing to 4096 (without changing the stack size), it's also unlikely that something will go wrong, but we'll crash (hard) if we do go past 1024.